configserver/src/main/java/com/yahoo/vespa/config/server/ApplicationRepository.java

```diff
@@ -426,12 +426,19 @@ public HttpResponse filedistributionStatus(ApplicationId applicationId, Duration
 
         Set<String> fileReferencesInUse = new HashSet<>();
         // Intentionally skip applications that we for some reason do not find
+        // or that we fail to get file references for (they will be retried on the next run)
         listApplications().stream()
                 .map(this::getOptionalApplication)
                 .map(Optional::get)
-                .forEach(application -> fileReferencesInUse.addAll(application.getModel().fileReferences().stream()
-                                                                           .map(FileReference::value)
-                                                                           .collect(Collectors.toSet())));
+                .forEach(application -> {
```

Comment:
I think this will still fail for the case we looked at earlier today (where `getOptionalApplication` is the method that throws, but it isn't called until `forEach` evaluates the stream calls).

How about rewriting it to a regular loop and wrap everything except `listApplications` in a try-catch?
```
for (var application : listApplications()) {
    try {
      ...
    } catch (Exception e) {
      ...
    }
}
```

Reply:
Yes, thanks, done.